datalad/core/local/configuration.py

```diff
@@ -0,0 +1,337 @@
+            variables) are considered according to the normal
+            rules of precedence. For action 'get' only 'dataset' and 'local'
+            (with include 'global' here) are supported. For action 'dump',
+            a scope selection is ignored and all scopes are considered.""",
+            constraints=EnsureChoice('global', 'local', 'dataset', None)),
+        spec=Parameter(
+            args=("spec",),
+            doc="""configuration name (for actions 'get' and 'unset'),
+            or name/value pair (for action 'set')""",
+            nargs='*',
+            metavar='name[=value]',
+            constraints=EnsureStr() | EnsureNone()),
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
+    )
+
+    @staticmethod
+    @datasetmethod(name='cfg')
+    @eval_results
+    def __call__(
+            action='dump',
+            spec=None,
+            scope=None,
+            dataset=None,
+            recursive=False,
+            recursion_limit=None):
+
+        # check conditions
+        # - global and recursion makes no sense
+
+        # normalize variable specificatons
+        specs = []
+        for s in ensure_list(spec):
+            if isinstance(s, tuple):
+                specs.append(s)
+            elif '=' not in s:
+                specs.append((s,))
+            else:
+                specs.append(tuple(s.split('=', 1)))
+
+        ds = None
+        if scope != 'global' or recursive:
+            ds = require_dataset(
+                dataset,
+                check_installed=True,
+                purpose='configuration')
+
+        res_kwargs = dict(
+            action='configuration',
+            logger=lgr,
+        )
+        if ds:
+            res_kwargs['refds'] = ds.path
+        yield from configuration(action, scope, specs, res_kwargs, ds)
+
+        if not recursive:
+            return
+
+        for subds in ds.subdatasets(
+                fulfilled=True,
+                recursive=True,
+                recursion_limit=recursion_limit,
+                on_failure='ignore',
+                result_renderer='disabled'):
+            yield from configuration(
+                action, scope, specs, res_kwargs, Dataset(subds['path']))
+
+    @staticmethod
+    def custom_result_renderer(res, **kwargs):
+        if (res['status'] != 'ok' or res['action'] != 'get_configuration'):
+            default_result_renderer(res)
+            return
+        # TODO source
+        from datalad.ui import ui
+        name = res['name']
+        if kwargs.get('action', None) == 'dump' and name in cfg_defs:
+            ui_def = cfg_defs[name].get('ui', [None, {}])[1]
+            for s in (ui_def.get('title'),
+                      ui_def.get('text')):
+                if s:
+                    ui.message('\n'.join(wrap(
+                        s,
+                        initial_indent='# ',
+                        subsequent_indent='# ',
```

Comment:
I think dumping this annotation very helpful, and I like the format.  I wonder if we should move the title/text extraction to `_dump`, letting the custom rerender just deal with the formatting.  That way the other renderers' output (I'm particularly interested in `--json{,-pp}`) would be more helpful.

Reply:
Yes, great idea!

Reply:
Done